### PR TITLE
feat(colormap): add colormap props and default properties

### PR DESCRIPTION
**Pull request by sedghi**
- Implementing colormap in stack viewport properties
- setColormap private + example
- api update
- vtk colomaps
- fix reset button + adding properties for volume
- added slabthickness to properties + update example
- small fix setColormapGPU
- fix voiRange for volume
- Mapping properties for volumeID
- small fix
- fix undefined properties
- properties by image index
- api updates
- refactoring VOI + about comments
- update imageIndex to imageId + doc changes
- update api's
- remove from basic example
- refacto + voiUpdateWithSetproperties back
- refactoring volume viewport properties
- re-add PTPrescaled
- fix
- fix documentation
- api checks
- try
- remove dummy file
- fix api check
- fix demo
- update examples
- api

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

- [] I have run the `yarn build:update-api` to update the API documentation, and have
  committed the changes to this PR. (Read more here https://www.cornerstonejs.org/docs/contribute/update-api)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS: 
- [] "Node version: 
- [] "Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

**Comment by salimkanoun** [comment]
Thanks for restauring this PR !

File: packages/core/src/RenderingEngine/BaseVolumeViewport.ts [hunk 1]
```diff
@@ -440,23 +453,56 @@ abstract class BaseVolumeViewport extends Viewport implements IVolumeViewport {
...
+   * Update the default properties for the volume viewport on the volume
+   * @param ViewportProperties - The properties to set
+   * @param volumeId The volume id to set the default properties for (if undefined, we set the global default viewport properties)
+   */
+  public setDefaultProperties(
+    ViewportProperties: VolumeViewportProperties,
+    volumeId?: string
+  ): void {
+    if (volumeId == null) {
+      this.globalDefaultProperties = ViewportProperties;
```
Comment (sedghi) [inline packages/core/src/RenderingEngine/BaseVolumeViewport.ts:475]: should'nt we return here if volumeId is null after we set to globalDefaults?

**Comment by sedghi** [comment]
@salimkanoun please check my recent three commits, and let me know if that looks fine to you as well
It basically enables reseting the colorbar to its initial values

**Comment by salimkanoun** [comment]
@sedghi sounds good to me !

**Pull request merged**